File: Infrastructure-Module/persistence/src/main/java/com/bamyanggang/persistence/common/exception/PersistenceException.java
package com.bamyanggang.persistence.common.exception;

import com.bamyanggang.commonmodule.exception.CustomException;

public class PersistenceException extends CustomException {
    public static final String CODE_PREFIX = "PERSISTENCE";

    public PersistenceException(int errorCode, String message) {
        super(CODE_PREFIX, errorCode, message);
    }

    public static class NotFound extends PersistenceException {
        public NotFound() {
            super(1, "해당 데이터를 찾을 수 없습니다.");
        }
    }

}

File: Infrastructure-Module/src/main/java/com/bamyanggang/infrastructuremodule/persistence/exception/PersistenceException.kt
package com.bamyanggang.infrastructuremodule.persistence.exception

import com.bamyanggang.commonmodule.exception.CustomException
import org.springframework.http.HttpStatus
import org.springframework.http.HttpStatusCode

sealed class PersistenceException(
    errorCode: Int,
    httpStatusCode: HttpStatusCode,
    message: String,
) : CustomException(CODE_PREFIX, errorCode, httpStatusCode , message) {

    class EntityNotFoundException(message: String = "Entity를 찾을 수 없습니다.") :
        PersistenceException(errorCode = 1, httpStatusCode = HttpStatus.NOT_FOUND, message = message)

    class EntityAlreadyExistsException(message: String = "Entity가 이미 존재합니다.") :
        PersistenceException(errorCode = 2, httpStatusCode = HttpStatus.CONFLICT, message = message)

    companion object {
        const val CODE_PREFIX = "PERSISTENCE"
    }

}